plugins/org/src/components/Cards/OwnershipCard/useGetEntities.ts

```diff
@@ -57,78 +57,129 @@ const getQueryParams = (
+    const childGroupEntities = (
+      await Promise.all(
+        childGroups.map(childGroup =>
+          limiter(() => catalogApi.getEntityByRef(childGroup)),
+        ),
+      )
+    ).filter(isEntity);
+
+    return (
+      await Promise.all(
+        childGroupEntities.map(childGroupEntity =>
+          limiter(() =>
+            getChildOwnershipEntityRefs(childGroupEntity, catalogApi),
+          ),
+        ),
+      )
+    ).flatMap(aggregated => aggregated);
   }
-  return owners;
+
+  return [
+    stringifyEntityRef({
+      kind: entity.kind,
+      namespace: entity.metadata.namespace,
+      name: entity.metadata.name,
+    }),
+  ];
 };
 
-const getAggregatedOwnersEntityRef = async (
-  parentGroup: Entity,
+const getOwners = async (
+  entity: Entity,
+  relationsType: AnyRelationsType,
   catalogApi: CatalogApi,
 ): Promise<string[]> => {
-  const requestedEntities: Entity[] = [];
-  const outstandingEntities = new Map<string, Promise<Entity | undefined>>();
-  const processedEntities = new Set<string>();
-  requestedEntities.push(parentGroup);
-  let currentEntity = parentGroup;
-
-  while (requestedEntities.length > 0) {
-    const childRelations = getEntityRelations(
-      currentEntity,
-      RELATION_PARENT_OF,
-      {
-        kind: 'Group',
-      },
-    );
+  const isGroup = entity.kind === 'Group';
+  const isAggregated = relationsType === 'aggregated';
+  const isUserEntity = entity.kind === 'User';
 
-    await Promise.all(
-      childRelations.map(childGroup =>
-        limiter(async () => {
-          const promise = catalogApi.getEntityByRef(childGroup);
-          outstandingEntities.set(childGroup.name, promise);
-          try {
-            const processedEntity = await promise;
-            if (processedEntity) {
-              requestedEntities.push(processedEntity);
-            }
-          } finally {
-            outstandingEntities.delete(childGroup.name);
-          }
-        }),
-      ),
-    );
-    requestedEntities.shift();
-    processedEntities.add(
-      stringifyEntityRef({
-        kind: currentEntity.kind,
-        namespace: currentEntity.metadata.namespace,
-        name: currentEntity.metadata.name,
-      }),
+  const owners: string[] = [];
+
+  if (isAggregated && isGroup) {
+    const childEntityRefs = await getChildOwnershipEntityRefs(
+      entity,
+      catalogApi,
     );
-    // always set currentEntity to the first element of array requestedEntities
-    currentEntity = requestedEntities[0];
+    owners.push(stringifyEntityRef(entity));
+    owners.push.apply(owners, childEntityRefs);
+  } else if (isAggregated && isUserEntity) {
+    const parentEntityRefs = getMemberOfEntityRefs(entity);
+    owners.push.apply(owners, parentEntityRefs);
+  } else {
+    owners.push(stringifyEntityRef(entity));
   }
 
-  return Array.from(processedEntities);
+  return owners;
 };
 
+export const DefaultRelationType = {
+  Direct: 'direct',
+  Aggregated: 'aggregated',
+} as const;
```

Comment:
This is to prevent TS from inferring the type to:
``` 
const DefaultRelationType: {
    Direct: string;
    Aggregated: string;
}
``` 
but rather the more useful:
```
const DefaultRelationType: {
    readonly Direct: "direct";
    readonly Aggregated: "aggregated";
}
```

which in turn infers `AnyRelationType` to `'direct' | 'aggregated'` instead of just `string`.